pkg/controller/logstorage/esgateway.go

```diff
@@ -47,7 +47,14 @@ func (r *ReconcileLogStorage) createEsGateway(
 		return reconcile.Result{}, false, nil
 	}
 
-	kubeControllersGatewaySecret, kubeControllersVerificationSecret, kubeControllersSecureUserSecret, err := common.CreateKubeControllersSecrets(ctx, esAdminUserSecret, r.client)
+	var esAdminUserName string
+	// This secret only ever contains one key.
```

Comment:
I think we should include a test to validate this assumption since we now depend on there only being a single key:value pair.

Reply:
I wasn't too sure how else to validate this assumption other than add a second key:value pair to a test es admin user secret and check to make sure that the correct (first) key:value pair is used as the username/password on the es-gateway deployment. Here is the change: https://github.com/tigera/operator/pull/1520/files#diff-569bac058c8112f0b2587062a0221cb55b557d6f071c32b4391354995c6571aeR452-R472

Let me know if that's what you had in mind @tmjd

Reply:
I was specifically thinking to include a check of `if len(esAdminUserSecret.Data) != 1` then Degrade here.

Reply:
I see, added the check